File: src/SELApproximator/SELApproximator.Tests/test/data/CSharpRecursiveFsa/ComplexOneLevelRec.cs
{caret}namespace N
{
    class Program
    {
        static void Main(string[] args)
        {
            var query0 = (1 > 0) ? "true" : "false";
            var query1 = "a";
            string query2 = Program.genString();
            string query3 = Program.genStringWithArg("d");
            query1 += query2;
            query1 = query1 + query3;
            SomeClass someClass = new SomeClass();
            for (int i = 0; i != 6; i++)
            {
                query1 += someClass.GenStrMethod();
            }
            string query5 = someClass.GenStrMethodWithArg("g" + query1);
            string query6 = someClass.GenStrMethodWithArgs(query5, query1);
            query6 += Program.withMultiReturn();
            Program.ExecuteImmediate(query6);
        }

        static void ExecuteImmediate(string query)
        {
            
        }

        static string genString()
        {
            var res = "b";
            return res;
        }

        static string genStringWithArg(string s)
        {
            var res = "c" + s;
            return res;
        }

        static string withMultiReturn()
        {
            string s = "y";
            var res = "";
            if (s.IsNormalized())
            {
                res = "x";
                return res;
            }
            res = s + "z";
            return res;
        }
    }

    public class SomeClass
    {
        public string GenStrMethod()
        {
            var res = "e";
            return res;
        }
        
        public string GenStrMethodWithArg(string s)
        {
            var res = "f" + s;
            return res;
        }

        public string GenStrMethodWithArgs(string s1, string s2)
        {
            var res = "h" + s1 + "j" + s2;
            return res;
        }
    }
}

File: src/SELApproximator/SELApproximator.Tests/test/data/JsRecursiveFsa/ComplexOneLevelRec.js
{caret}function namespace()
{
    function genString() 
    {
        var r = "b";
        return r;
    }

    function genStringWithArg(s)
    {
        var r = "c" + s;
        return r;
    }

    function withMultiReturn()
    {
        var s = "y";
        var r = "x";
        if (s.IsNormalized())
        {
            return r;
        }
        r = s + "z";
        return r;
    }

    function GenStrMethod()
    {
        var res = "e";
        return res;
    }
        
    function GenStrMethodWithArg(s)
    {
        var res = "f" + s;
        return res;
    }

    function GenStrMethodWithArgs(s1, s2)
    {
        var res = "h" + s1 + "j" + s2;
        return res;
    }

    function main(args)
    {
        var query0 = (1 > 0) ? "true" : "false";
        var query1 = "a";
        var query2 = genString();
        var query3 = genStringWithArg("d");
        query1 += query2;
        query1 = query1 + query3;
        for (var i = 0; i < 6; i++)
        {
            query1 += GenStrMethod();
        }
        var query5 = GenStrMethodWithArg("g" + query1);
        var query6 = GenStrMethodWithArgs(query5, query1);
        query6 += withMultiReturn();
        execScript(query6);
    }
}